jlm/hls/ir/hls.hpp

```diff
@@ -1093,39 +1100,26 @@ class mem_req_op final : public jlm::rvsdg::simple_op
   }
 
   mem_req_op(
-      const std::vector<const rvsdg::valuetype *> & load_types,
-      const std::vector<const rvsdg::valuetype *> & store_types)
+      const std::vector<std::shared_ptr<const rvsdg::valuetype>> & load_types,
+      const std::vector<std::shared_ptr<const rvsdg::valuetype>> & store_types)
       : simple_op(CreateInTypes(load_types, store_types), CreateOutTypes(load_types, store_types))
   {
     LoadTypes_ = new std::vector<std::shared_ptr<const rvsdg::type>>();
     StoreTypes_ = new std::vector<std::shared_ptr<const rvsdg::type>>();
     for (auto loadType : load_types)
     {
       JLM_ASSERT(
-          dynamic_cast<const rvsdg::bittype *>(loadType)
-          || dynamic_cast<const llvm::PointerType *>(loadType));
-      LoadTypes_->emplace_back(loadType->copy());
+          dynamic_cast<const rvsdg::bittype *>(loadType.get())
+          || dynamic_cast<const llvm::PointerType *>(loadType.get()));
```

Comment:
`JLM_ASSERT(is<rvsdg::bittype*>(loadType.get()) || is<llvm::PointerType>(loadedType.get()));`

Reply:
that does not compile -- I cannot find definition for something named "is"

Reply:
oh right now I remember that I even touched it -- fixed